### PR TITLE
feat(component_interface_utils): apply message change

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Apply message change https://github.com/autowarefoundation/autoware.universe/pull/1897 to component_interface_utils package.
Use template functions for response status.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/component_interface_utils/include/component_interface_utils/status.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// limitations under the License.
+
+#ifndef COMPONENT_INTERFACE_UTILS__STATUS_HPP_
+#define COMPONENT_INTERFACE_UTILS__STATUS_HPP_
+
+namespace component_interface_utils::status
+{
+
+template <class T1, class T2>
+void copy(const T1 & src, T2 & dst)  // NOLINT cpplint false positive
```
Comment (kenji-miyake) [inline common/component_interface_utils/include/component_interface_utils/status.hpp:22]: Could you write the error message here?

**Review by kenji-miyake — APPROVED** [review]

File: common/component_interface_utils/include/component_interface_utils/status.hpp (hunk 1 above)
Comment (isamu-takagi) [reply common/component_interface_utils/include/component_interface_utils/status.hpp]: Is it like this https://github.com/autowarefoundation/autoware.universe/pull/1921/commits/1c5b53ba94bfe272c67df8c7798f8ee7690e7da3? The error message is:
```
Add #include <algorithm> for copy  [build/include_what_you_use] [4]
```

Comment (isamu-takagi) [reply common/component_interface_utils/include/component_interface_utils/status.hpp]: In addition, this also causes false positives in execution.
```
  component_interface_utils::status::copy(a, b);  // cpplint requires algorithm header
```

Comment (kenji-miyake) [reply common/component_interface_utils/include/component_interface_utils/status.hpp]: Ah, no. I just wanted to know this one. Thanks!

```
Add #include <algorithm> for copy  [build/include_what_you_use] [4]
```

Comment (kenji-miyake) [reply common/component_interface_utils/include/component_interface_utils/status.hpp]: > In addition, this also causes false positives in execution.

Maybe related? https://github.com/cpplint/cpplint/issues/200

**Pull request merged**